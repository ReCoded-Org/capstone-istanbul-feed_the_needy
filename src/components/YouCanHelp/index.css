.containerYoucanhelp {
  margin-bottom: 5em;
}

.youCanHelpToo {
  display: flex;
  justify-content: center;
}

.imageSection {
  width: 20em;
  height: 30em;
  display: flex;
  justify-content: center;
}

.foodImage > .ant-image-img {
  width: 100%;
  display: flex;
  justify-content: center;
  border-bottom-left-radius: 10em;
  border-top-left-radius: 10em;
}

.ant-col.ant-col-16.imageSection.ant-col-xs-24.ant-col-sm-20.ant-col-md-24.ant-col-xl-10 {
  height: min-content;
}

.youCanHelpText {
  width: 40em;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  text-align: center;
  margin: 0 3em 1em 3em;
  padding-top: 3em;
}

.youCanHelpText span {
  font-size: 1.5em;
}

.youCanHelpText h1 {
  font-size: 3em;
  text-shadow: 0.1em 0.1em rgb(219, 212, 212);
}

.donateButton {
  width: 15em;
  height: 3em;
  margin-top: 3em;
  margin-bottom: 3em;
  background-color: #ff3c21;
  color: #ffffff;
}
