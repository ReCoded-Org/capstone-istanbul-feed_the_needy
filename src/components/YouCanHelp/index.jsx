import React from "react";
import HelpImage from "./image.png";
import { Button, Row, Col, Image } from "antd";
import { useTranslation } from "react-i18next";
import "antd/dist/antd.css";
import "./index.css";
import { useHistory } from "react-router-dom";

const YouCanHelp = ({ isTesting }) => {
  const { t } = useTranslation();
  const history = useHistory();

  return (
    <div className="containerYoucanhelp">
      <Row span={36} xs={24} sm={20} md={36} xl={30} justify="space-between">
        <Col
          className="youCanHelpToo"
          span={16}
          xs={24}
          sm={20}
          md={24}
          xl={10}
        >
          <Col className="youCanHelpText">
            <h1>You can help too!</h1>
            <span>{t("youcanhelp.headers.textSection1")}</span>
            <Col>
              {isTesting ? (
                <button>Donate</button>
              ) : (
                <Button
                  onClick={() => history.push("/coupons")}
                  className="donateButton"
                >
                  {t("youcanhelp.headers.button")}
                </Button>
              )}
            </Col>
          </Col>
        </Col>
        <Col className="imageSection" span={16} xs={24} sm={20} md={24} xl={10}>
          <Image
            className="foodImage"
            xs={20}
            sm={20}
            md={10}
            xl={10}
            src={HelpImage}
            alt="help"
          />
        </Col>
      </Row>
    </div>
  );
};

export default YouCanHelp;
