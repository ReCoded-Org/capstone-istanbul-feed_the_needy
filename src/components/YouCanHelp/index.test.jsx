import React from "react";
import renderer from "react-test-renderer";
import YouCanHelp from ".";

jest.mock("i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

const isTesting = true;

it("renders correctly", () => {
  const tree = renderer.create(<YouCanHelp isTesting={isTesting} />).toJSON();
  expect(tree).toMatchSnapshot();
});
