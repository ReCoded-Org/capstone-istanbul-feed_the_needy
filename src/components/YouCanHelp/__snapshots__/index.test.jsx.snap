// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="containerYoucanhelp"
>
  <div
    className="ant-row ant-row-space-between"
    md={36}
    sm={20}
    span={36}
    style={Object {}}
    xl={30}
    xs={24}
  >
    <div
      className="ant-col ant-col-16 youCanHelpToo ant-col-xs-24 ant-col-sm-20 ant-col-md-24 ant-col-xl-10"
      style={Object {}}
    >
      <div
        className="ant-col youCanHelpText"
        style={Object {}}
      >
        <h1>
          youcanhelp.headers.title
        </h1>
        <span>
          youcanhelp.headers.textSection1
        </span>
        <div
          className="ant-col"
          style={Object {}}
        >
          <button>
            Donate
          </button>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-16 imageSection ant-col-xs-24 ant-col-sm-20 ant-col-md-24 ant-col-xl-10"
      style={Object {}}
    >
      <div
        className="ant-image foodImage"
        md={10}
        onClick={[Function]}
        sm={20}
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
        xl={10}
        xs={20}
      >
        <img
          alt="help"
          className="ant-image-img"
          onError={[Function]}
          onLoad={[Function]}
          src="image.png"
        />
      </div>
    </div>
  </div>
</div>
`;
